### PR TITLE
[Test] Equivalence tests for `evaluate`, `execute` and `finalize`

### PR DESCRIPTION
**Pull request by d0cd**
This PR adds tests checking that the results of `evaluate`, `execute` and `finalize` for `assert`, `is`, `commit`, and `hash` instructions are consistent/equivalent.

### CODE REVIEW

File: synthesizer/src/program/instruction/operation/cast.rs [hunk 1]
```diff
@@ -526,6 +445,60 @@ impl<N: Network> Cast<N> {
...
+        // Initialize the struct members.
+        let mut members = IndexMap::new();
+        for (member, (member_name, member_type)) in inputs.iter().zip_eq(struct_.members()) {
+            // Compute the register type.
+            let register_type = RegisterType::Plaintext(*member_type);
+            // Retrieve the plaintext value from the entry.
+            let plaintext = match member {
+                Value::Plaintext(plaintext) => {
+                    // Ensure the member matches the register type.
+                    stack.matches_register_type(&Value::Plaintext(plaintext.clone()), &register_type)?;
```
Comment (ljedrz) [inline synthesizer/src/program/instruction/operation/cast.rs]: it's not introduced by this PR, but we can avoid this clone by using the publicly available `Stack::matches_plaintext`

File: synthesizer/src/program/instruction/operation/commit.rs [hunk 2]
```diff
@@ -285,9 +285,269 @@ impl<N: Network, const VARIANT: u8> ToBytes for CommitInstruction<N, VARIANT> {
...
+                    let operation = |operands, destination| $commit::<CurrentNetwork> { operands, destination };
+                    // Initialize the opcode.
+                    let opcode = $commit::<CurrentNetwork>::opcode();
+
+                    // Prepare the rng.
+                    let mut rng = TestRng::default();
+
+                   // Prepare the test.
+                    let modes_a = [/* circuit::Mode::Constant, */ circuit::Mode::Public, circuit::Mode::Private];
+                    let modes_b = [/* circuit::Mode::Constant, */ circuit::Mode::Public, circuit::Mode::Private];
```
Comment (ljedrz) [inline synthesizer/src/program/instruction/operation/commit.rs:440]: perhaps we should add a TODO on when these can be uncommented? (applies to all such instances)

**Review by ljedrz — COMMENTED** [review]
Left a couple of comments.

**Comment by d0cd** [comment]
Closed in favor of PRs: 1532-1537

**Pull request closed without merging**